test/e2e/apimachinery/webhook.go

```diff
@@ -788,10 +788,11 @@ func testWebhookForWebhookConfigurations(f *framework.Framework) {
 				Name: "should-be-removable-validating-webhook.k8s.io",
 				Rules: []v1beta1.RuleWithOperations{{
 					Operations: []v1beta1.OperationType{v1beta1.Create},
+					// This will not match any real resources so this webhook should never be called.
 					Rule: v1beta1.Rule{
-						APIGroups:   []string{"*"},
-						APIVersions: []string{"*"},
-						Resources:   []string{"*"},
+						APIGroups:   []string{""},
+						APIVersions: []string{"v1"},
+						Resources:   []string{"invalid"},
```

Comment:
These is actually one configuration already doing that, these ones are there to test if a hook can prevent deleting another hook's configuration (to prevent a cycle of hooks preventing each other's deletions)

Reply:
https://github.com/jennybuckley/kubernetes/blob/cd06e4db1fa8c2eda0e08d8699964fa41fca3cfa/test/e2e/apimachinery/webhook.go#L782-L813

Reply:
> these ones are there to test if a hook can prevent deleting another hook's configuration

I don't see how selecting an invalid resource tests that case

Reply:
These webhook configurations should be able to be registered and deleted without interference from the webhook I linked above. If webhooks are able to prevent deletion of other webhook configurations, the webhook I linked above will cause [this check](https://github.com/jennybuckley/kubernetes/blob/cd06e4db1fa8c2eda0e08d8699964fa41fca3cfa/test/e2e/apimachinery/webhook.go#L822) to fail, which it did, as shown in this test result from #59840 where just the test commit was merged and not the fix:
```
test/e2e/apimachinery/webhook.go:147
deleting webhook config e2e-test-should-be-removable-validating-webhook-config with namespace e2e-tests-webhook-qlmqf
Expected error:
    <*errors.StatusError | 0xc421454b40>: {
        ErrStatus: {
            TypeMeta: {Kind: "", APIVersion: ""},
            ListMeta: {SelfLink: "", ResourceVersion: "", Continue: ""},
            Status: "Failure",
            Message: "admission webhook \"deny-webhook-configuration-deletions.k8s.io\" denied the request: this webhook denies all requests",
            Reason: "",
            Details: nil,
            Code: 500,
        },
    }
    admission webhook "deny-webhook-configuration-deletions.k8s.io" denied the request: this webhook denies all requests
not to have occurred
test/e2e/apimachinery/webhook.go:815
```

Reply:
The test goes like this:
1. Create a [webhook](https://github.com/jennybuckley/kubernetes/blob/cd06e4db1fa8c2eda0e08d8699964fa41fca3cfa/test/e2e/apimachinery/webhook.go#L737-L748) which is configured to prevent deletion of ValidatingWebhookConfiguration and MutatingWebhookConfiguration objects
2. Attempt to create and delete a dummy ValidatingWebhookConfiguration](https://github.com/jennybuckley/kubernetes/blob/cd06e4db1fa8c2eda0e08d8699964fa41fca3cfa/test/e2e/apimachinery/webhook.go#L782-L813)
3. Attempt to create and delete a dummy [MutatingWebhookConfiguration](https://github.com/jennybuckley/kubernetes/blob/cd06e4db1fa8c2eda0e08d8699964fa41fca3cfa/test/e2e/apimachinery/webhook.go#L826-L857)

Reply:
I see, thanks.